### PR TITLE
fix: TECH-522 rework of redux offline retry handling

### PR DESCRIPTION
**Pull request by JoakimSM**
- reworked fix for redux offline retry handling after moving to app runtime dataengine

### CODE REVIEW

File: src/core_modules/capture-core/flow/app-runtime_v2.x.x.js [hunk 1]
```diff
@@ -0,0 +1,121 @@
+declare module '@dhis2/app-runtime' {
```
Comment (JoakimSM) [inline src/core_modules/capture-core/flow/app-runtime_v2.x.x.js:1]: New code in this file is only line 97 - 120. See next comment.

File: src/core_modules/capture-core/trackerOffline/getEffectReconciler.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+// @flow
```
Comment (JoakimSM) [inline src/core_modules/capture-core/trackerOffline/getEffectReconciler.js:1]: This file is old code. Moved to its own file.

File: src/core_modules/capture-core/trackerOffline/queueConfig.js [hunk 3]
```diff
@@ -1,20 +1,6 @@
 // @flow
```
Comment (JoakimSM) [inline src/core_modules/capture-core/trackerOffline/queueConfig.js:1]: This is old code, extracted `queueConfig` to its own file.

File: .eslintrc [hunk 4]
```diff
@@ -43,7 +43,7 @@
                 "ignorePattern": "^\\s*var\\s.+=\\s*require\\s*\\("
             }
         ],
-        "complexity": [1, 5],
+        "complexity": [1, 8],
```
[the remark this answers is not in the document]
Comment (paschalidi) [reply .eslintrc:46]: 8 looks good to me. I also have problems sometimes with this complexity warning

File: src/core_modules/capture-core/trackerOffline/shouldDiscard.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    const statusCodeSearchResult = message.match(/\(\d*\)$/);
+    if (statusCodeSearchResult) {
+        const statusCode = statusCodeSearchResult[0].replace(/[()]/g, '');
+        return Boolean(statusCode) && ![408, 429, 502, 503, 504].includes(statusCode);
+    }
+    return false;
+};
+
+export const shouldDiscard = (error?: ApiError) => {
+    if (!error || !error.type) {
```
Comment (paschalidi) [inline src/core_modules/capture-core/trackerOffline/shouldDiscard.js:22]: just curious if you could do something like
```suggestion
    if (!error?.type) {
```

Comment (JoakimSM) [reply src/core_modules/capture-core/trackerOffline/shouldDiscard.js]: Nice! That should work.

**Pull request merged**